pymeasure/instruments/attocube/anc300.py

```diff
@@ -0,0 +1,227 @@
+    """ get_process function for the Attocube console to obtain a float from
+    the reply
+
+    :param reply: reply string
+    :returns: string with only the numerical value
+    """
+    return float(extract_value(reply))
+
+
+def extract_int(reply):
+    """ get_process function for the Attocube console to obtain an integer from
+    the reply
+
+    :param reply: reply string
+    :returns: string with only the numerical value
+    """
+    return int(extract_value(reply))
+
+
+class Axis(object):
+    """ Represents a single open loop axis of the Attocube ANC350
+
+    :param axis: axis identifier, integer from 1 to 7
+    :param controller: ANC300Controller instance used for the communication
+    """
+
+    serial_nr = Instrument.measurement("getser",
+                                       "Serial number of the axis")
+
+    voltage = Instrument.control(
+            "getv", "setv %.3f",
+            """ Amplitude of the stepping voltage in volts from 0 to 150 V. This
+            property can be set. """,
+            validator=strict_range, values=[0, 150],
+            get_process=extract_float)
+
+    frequency = Instrument.control(
+            "getf", "setf %.3f",
+            """ Frequency of the stepping motion in Hertz from 1 to 10000 Hz.
+            This property can be set. """,
+            validator=strict_range, values=[1, 10000],
+            get_process=extract_int)
+
+    mode = Instrument.control(
+            "getm", "setm %s",
+            """ Axis mode. This can be 'gnd', 'inp', 'cap', 'stp', 'off',
+            'stp+', 'stp-'. Available modes depend on the actual axis model""",
+            validator=strict_discrete_set,
+            values=['gnd', 'inp', 'cap', 'stp', 'off', 'stp+', 'stp-'],
+            get_process=extract_value)
+
+    offset_voltage = Instrument.control(
+            "geta", "seta %.3f",
+            """ Offset voltage in Volts from 0 to 150 V.
+            This property can be set. """,
+            validator=strict_range, values=[0, 150],
+            get_process=extract_float)
+
+    output_voltage = Instrument.measurement("geto",
+                                            """ Output voltage in volts.""",
+                                            get_process=extract_float)
+
+    capacity = Instrument.measurement("getc",
+                                      """ Saved capacity value in nF of the
+                                      axis.""",
+                                      get_process=extract_float)
+
+    stepu = Instrument.setting(
+            "stepu %d",
+            """ Step upwards for N steps. Mode must be 'stp' and N must be
+            positive.""",
+            validator=strict_range, values=[0, inf])
+
+    stepd = Instrument.setting(
+            "stepd %d",
+            """ Step downwards for N steps. Mode must be 'stp' and N must be
+            positive.""",
+            validator=strict_range, values=[0, inf])
+
+    def __init__(self, axis, controller):
```

Comment:
please put the "parent/instrument/controller" first, then the name of the axis.

Reply:
ok